Documentation additions

danforbes opened the pull request:
Going to do this piecemeal because it will get out of control quickly.

philpax commented on crates/ggml/src/lib.rs:
```diff
@@ -29,11 +29,11 @@ pub type ElementType = Type;
 /// The format of the file containing the model.
 pub enum ContainerType {
-    /// `GGML`: legacy format, oldest ggml tensor file format
+    /// `Legacy format, oldest ggml tensor file format
```
Accidental backtick left here

philpax commented on crates/llm-base/src/lib.rs:
```diff
@@ -32,18 +32,17 @@ pub use vocabulary::{TokenBias, TokenId, Vocabulary};
     pub n_threads: usize,
-    /// [InferenceSession::feed_prompt] processes the prompt in batches of tokens.
-    /// This controls how large an individual batch is.
+    /// Controls batch size for [InferenceSession::feed_prompt].
```
Explain what a batch is here

philpax commented on crates/llm/src/lib.rs:
```diff
@@ -1,8 +1,61 @@
+//! - [GPT-2](https://huggingface.co/docs/transformers/model_doc/gpt2)
+//! - [GPT-J](https://huggingface.co/docs/transformers/model_doc/gptj)
+//! - [LLaMA](https://huggingface.co/docs/transformers/model_doc/llama)
+//! - [GPT-NeoX](https://huggingface.co/docs/transformers/model_doc/gpt_neox)
```
I'd suggest linking to the crates here (e.g. https://crates.io/crates/llm-neox), as you already link to HF in each crate's doc

philpax commented:
LGTM, apart from the above 🚀 ! I'd suggest adding `#![deny(missing_docs)]` to each of the crates if not already present, so that the compiler will tell you what needs to be documented.

danforbes commented:
I've addressed all your comments (I hope) and added a few extra docs...I'd love some help filling in the docs for the hyperparamaters that I wasn't able to do myself. Please feel free to make any changes you'd like and merge this when you see fit. I will continue working on docs in separate PRs to keep things manageable.

philpax commented:
Looks good to me. Yeah, the hyperparameters are... not the most descriptive, I didn't document them because I wasn't sure myself. I'll have to get back to that at some point.

I'll merge this, and we can keep iterating on it. Great work so far!
